native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.android.js

```diff
@@ -0,0 +1,241 @@
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import BpkText from 'react-native-bpk-component-text';
+import BpkAnimateHeight from 'react-native-bpk-component-animate-height';
+import BpkIcon from 'react-native-bpk-component-icon';
+
+import { dismissablePropType } from './customPropTypes';
+
+import AnimateAndFade from './AnimateAndFade';
+
+export const ALERT_TYPES = {
+  SUCCESS: 'success',
+  WARN: 'warn',
+  ERROR: 'error',
+  NEUTRAL: 'neutral',
+};
+
+const styles = StyleSheet.create({
+  mainContainer: {
+    paddingVertical: spacingBase,
+    paddingHorizontal: spacingLg,
+    backgroundColor: colorGray50,
+    minHeight: spacingXxl + spacingBase,
+    flex: 1,
+  },
+  dismissableContainer: {
+    paddingVertical: spacingBase,
+    paddingHorizontal: spacingLg,
+    backgroundColor: colorGray50,
+  },
+  childrenContainer: {
+    paddingVertical: spacingSm,
+  },
+  container: {
+    display: 'flex',
+    flexDirection: 'row',
+  },
+  actionContainer: {
+    alignItems: 'flex-end',
+    justifyContent: 'center',
+    paddingHorizontal: spacingLg,
+    backgroundColor: colorGray50,
+  },
+  text: {
+    flex: 1,
+  },
+  icon: {
+    paddingRight: spacingSm,
+    paddingTop: spacingSm - spacingSm / spacingSm,
+  },
+  dismiss: {
+    color: colorBlue500,
+  },
+});
+
+const ALERT_TYPE_STYLES = {
+  [ALERT_TYPES.SUCCESS]: {
+    iconSource: 'tick-circle',
+    iconStyle: {
+      color: colorGreen500,
+    },
+  },
+  [ALERT_TYPES.WARN]: {
+    iconSource: 'information-circle',
+    iconStyle: {
+      color: colorYellow500,
+    },
+  },
+  [ALERT_TYPES.ERROR]: {
+    iconSource: 'information-circle',
+    iconStyle: {
+      color: colorRed500,
+    },
+  },
+  [ALERT_TYPES.NEUTRAL]: {
+    iconSource: 'information-circle',
+    iconStyle: {
+      color: colorGray500,
+    },
+  },
+};
+
+const BpkBannerAlert = props => {
+  const {
+    type,
+    message,
+    onDismiss,
+    onToggleExpanded,
+    dismissable,
+    expanded,
+    dismissButtonLabel,
+    toggleExpandedButtonLabel,
+    show,
+    style: userStyle,
+    animateOnEnter,
+    animateOnLeave,
+    children,
+    ...rest
+  } = props;
+
+  const expandable = children !== null;
+  const { iconSource, iconStyle } = ALERT_TYPE_STYLES[type] || {};
+
+  const dismissableContent = (
+    <TouchableNativeFeedback
+      accessibilityComponentType="button"
+      accessibilityLabel="dismiss"
+      background={TouchableNativeFeedback.SelectableBackgroundBorderless()}
+      useForeground
+      onPress={onDismiss}
+    >
+      <View style={styles.actionContainer}>
+        <BpkText textStyle="sm" emphasize style={styles.dismiss}>
+          DISMISS
+        </BpkText>
+      </View>
+    </TouchableNativeFeedback>
+  );
+  const expandableContent = (
+    <BpkAnimateHeight expanded={expanded}>
+      <View style={styles.childrenContainer}>{children}</View>
+    </BpkAnimateHeight>
+  );
+  const expandableIcon = (
```

Comment:
This and `dismissableContent` feels pretty componenty. Should they be components?